enacts/crop_suitability/maproom_crop_suit.py

```diff
@@ -0,0 +1,548 @@
+            ),
+            name="Street",
+            checked=False,
+        ),
+        dlf.BaseLayer(
+            dlf.TileLayer(
+                url="https://{s}.tile.opentopomap.org/{z}/{x}/{y}.png"
+            ),
+            name="Topo",
+            checked=True,
+        ),
+    ] + [
+        make_adm_overlay(
+            adm["name"],
+            adm["sql"],
+            adm["color"],
+            i+1,
+            len(GLOBAL_CONFIG["shapes_adm"])-i,
+            is_checked=adm["is_checked"]
+        )
+        for i, adm in enumerate(GLOBAL_CONFIG["shapes_adm"])
+    ] + [
+        dlf.Overlay(
+            dlf.TileLayer(
+                url=f"{TILE_PFX}/{{z}}/{{x}}/{{y}}?{qstr}",
+                opacity=1,
+            ),
+            name="Crop Suitability",
+            checked=True,
+        ),
+    ]
+
+
+@APP.callback(
+    Output("navbar-collapse", "is_open"),
+    Input("navbar-toggler", "n_clicks"),
+    State("navbar-collapse", "is_open"),
+)
+def toggle_navbar_collapse(n, is_open):
+    if n:
+        return not is_open
+    return is_open
+
+
+@APP.callback(
+    Output("hover_feature_label", "children"),
+    Input({"type": "borders_adm", "index": ALL}, "hover_feature")
+)
+def write_hover_adm_label(adm_loc):
+    location_description = "the map will return location name"
+    for i, adm in enumerate(adm_loc):
+        if adm is not None:
+            location_description = adm['geometry']['label']
+    return f'Mousing over {location_description}'
+
+
+@APP.callback(
+    Output("map_description", "children"),
+    Input("data_choice", "value"),
+)
+def write_map_description(data_choice):
+    return CONFIG["map_text"][data_choice]["description"]    
+
+@APP.callback(
+    Output("loc_marker", "position"),
+    Output("lat_input", "value"),
+    Output("lng_input", "value"),
+    Input("submit_lat_lng","n_clicks"),
+    Input("map", "click_lat_lng"),
+    State("lat_input", "value"),
+    State("lng_input", "value")
+)
+def pick_location(n_clicks, click_lat_lng, latitude, longitude):
+    if dash.ctx.triggered_id == None:
+        lat = rr_mrg.precip["Y"][int(rr_mrg.precip["Y"].size/2)].values
+        lng = rr_mrg.precip["X"][int(rr_mrg.precip["X"].size/2)].values
+    else:
+        if dash.ctx.triggered_id == "map":
+            lat = click_lat_lng[0]
+            lng = click_lat_lng[1]
+        else:
+            lat = latitude
+            lng = longitude
+        try:
+            nearest_grid = pingrid.sel_snap(rr_mrg.precip, lat, lng)
+            lat = nearest_grid["Y"].values
+            lng = nearest_grid["X"].values
+        except KeyError:
+            lat = lat
+            lng = lng
+    return [lat, lng], lat, lng
+
+
+def crop_suitability(
+    rainfall_data,
+    min_wet_days,
+    wet_day_def,
+    tmax_data,
+    tmin_data,
+    lower_wet_threshold,
+    upper_wet_threshold,
+    max_temp,
+    min_temp,
+    temp_range,
+    target_season,
+):
+    if isinstance(rainfall_data,xr.DataArray):
```

Comment:
I don't understand this if/else block. I think one case is where it's called for the timeseries plot and the other is for the map, but I don't see any reason it should be a Dataset in one case and a DataArray in the other--that just seems like a coincidence that happens to hold in the current version but might not in the future. Also, I don't understand why you need to select out data from the target season in one case but not in the other. Wouldn't you need to do that in both cases?

Reply:
Yes, I think it would be good to do the season selection in one place that's on the path for both map and time series. Then in the map case we select a single year, and in the time series we select a single point.

> The difference in the data structure is a result of the timeseries data being one lat/lng for all timesteps versus the tile being all lat/lng for one point in time. 

There's nothing about those two array shapes that dictates that one must be a Dataset and the other a DataArray. Both xarray classes can represent both shapes. There's just some inessential detail of the current implementation that makes one of them come out a Dataset and the other a DataArray. If you needed to know the array's shape, you should look directly at its shape (check how many dimensions it has, check if it has a dimension called `T`, ...), not at whether it's stored in a Dataset or a DataArray. But if you move the selection of the season out of this function as suggested above, then I don't think you even need to know the shape anymore. The `if` case and the `else` case can be collapsed into one.

Reply:
That sounds like a bug in `sel_snap`--if given a Dataset, it should return a Dataset. I'll file an issue for that. But in any case, I don't think it makes any difference for this application.  Most operations that can be applied to a Datset can also be applied to a DataArray and vice versa.

Reply:
> If that is not good practice I can make a new empty dataset to append the outputs to.

Yeah, that seems more robust to future changes.

Reply:
Okay, I changed it so that now I create an empty dataset like this:

    crop_suitability = xr.Dataset(
        data_vars = dict(
        ),
        coords = dict(
            X = avg_tmax["X"],
            Y = avg_tmax["Y"],
            year = avg_tmax["year"],
        ),
    )

As of now I am still having a hard time figuring out a clean way to append data to an empty dataset in both cases of appending from a dataArray and dataset, since the way you can select variables and data from within the two is different.

I will keep looking but right now the easiest thing I think to do is to convert the dataarrays to datasets before passing them into the crop_suitability() function.

For example:

    crop_suitability = crop_suitability.assign(
        max_temp = tmax[temp_var], min_temp = tmin[temp_var],
        temp_range = avg_temp_range[temp_var], precip_range = total_precip_range[precip_var],
        wet_days = wet_days[precip_var])

Where tmax, tmin, etc are dataset/datarrays with outputs from calculations done on the original data

Trying to assign data to the variables in the new dataset using this type of indexing works for a dataset, but since the dataarray does not use variable names in the same way, it does not let you select for the data in the same way.. `tmax[temp_var]` only works for the dataset; but simply using `tmax` to call the data would not work for the dataset.

Maybe there is an easy workaround I am not seeing for this?

Reply:
update: I pushed the edited code for you to look at

Reply:
I looked at the new version and frankly I can't make heads or tails of it. At [one point](https://github.com/iridl/python-maprooms/pull/280/files#diff-329e8bc1434c84455197fee53abce5eb2b89e5be9cdc5b63434cc6c4879ec297R308-R311) you seem to be taking the sum of max temp, min temp, temp range, precip range, and wet days?! I think you're better off discarding your most recent changes and going back to the version this comment was on.

I gave bad advice before. Datasets and DataArrays are interchangeable when you're doing calculations on them, but they're not interchangeable when you need to get the results out. I think the root of the problem here is actually that `rr_mrg`, `tmin_mrg`, and `tmax_mrg` are Datasets when they ought to be DataArrays. In all the other maprooms they're DataArrays, see e.g. (monthly)[https://github.com/iridl/python-maprooms/blob/master/enacts/monthly/maproom.py#L59], where the `[GLOBAL_CONFIG['dekadal']['vars'][name][2]]` is the part that extracts the DataArray from the Dataset. That part is missing from e.g. [line 41](https://github.com/iridl/python-maprooms/pull/280/files#diff-329e8bc1434c84455197fee53abce5eb2b89e5be9cdc5b63434cc6c4879ec297R41) of maproom_crop_suit.py. Once everything is a DataArray, you should have an easier time combining the two cases.

BTW I don't think there's a bug in `sel_snap` after all. Every time you call `sel_snap` you're giving it a DataArray (e.g. `rr_mrg.precip`), not a Dataset (`rr_mrg`), so returning a DataArray is the expected behavior.

Reply:
The part of the crop_suitability() function you are referring to where I sum all of those variables is creating the index - perhaps what I name each of those variables in the dataset is midleading. What they are are just '0' or '1' if the condition was met in a prior calculating in the function. So it is adding this up to get a number 1-5 where 5 has all of the conditions met and 0 has none of the conditions met. Not actually just adding up temperature, precip, etc values. This was in the previous commit. 

Yes I did not notice how I did not select the DataArray at the beginning - this fixes my issue! I will push a commit with this improvement.